pkg/depend/write.go

```diff
@@ -1,12 +1,75 @@
 package depend
 
-import "github.com/git-depend/git-depend/pkg/git"
+import (
+	"encoding/json"
+	"errors"
+	"time"
 
-// Write a request to a note.
-func (req *Request) Write(cache *git.Cache, ref string) error {
-	json, err := req.GetJson()
+	"github.com/git-depend/git-depend/pkg/git"
+	"github.com/git-depend/git-depend/pkg/utils"
+)
+
+type Status int
+
+const (
+	Locked Status = iota
+	Unlocked
+)
+
+func (s Status) String() string {
+	return [...]string{"Locked", "Unlocked"}[s]
+}
+
+var ref_lock_name string = "git-depend-lock"
+
+// Lock allows us to safely write to a note.
+type Lock struct {
+	ID        string    `json:"Id"`
+	Timestamp time.Time `json:"Timestamp"`
+	Status    string    `json:"Status"`
+}
+
+// WriteLock will lock an individual repository.
+func (lock *Lock) WriteLock(cache *git.Cache, node *Node) error {
+	data, err := json.Marshal(lock)
 	if err != nil {
 		return err
 	}
-	return cache.ForceAddNotes(req.Url, ref, string(json))
+
+	if err = cache.AddNotes(node.URL, ref_lock_name, string(data)); err != nil {
+		return err
+	}
+
+	return cache.PushNotes(node.URL, ref_lock_name)
+}
+
+// WriteLocks for a request and the children.
+func (requests *Requests) WriteLocks(cache *git.Cache) error {
```

Comment:
Should this be in `request.go` instead?

Reply:
I was considering splitting out the write stuff from the request stuff because I think the collision logic for write might take up quite a bit of space. It's a little annoying the way it's written currently because the request also creates the `Lock` in that function too. I might leave this for now and we can think about it in the future.